### PR TITLE
Mobile App: loading when fetch threads

### PR DESCRIPTION
**Pull request by loi-huynhphuc**
Mobile App: loading when fetch threads https://github.com/nccasia/mezon/issues/4763

https://github.com/user-attachments/assets/fc9be415-1c02-42ea-9d93-e1a1a89fd6e0

### CODE REVIEW

File: apps/mobile/src/app/components/ThreadDetail/index.tsx [hunk 1]
```diff
@@ -42,20 +44,25 @@ export default function CreateThreadModal({ navigation, route }: MenuThreadScree
...
-				clanId: currentChannel?.clan_id ?? '',
-				noCache: true
-			};
-			await dispatch(threadsActions.fetchThreads(body));
-		};
-
 		fetchThreads();
+		return () => {
+			dispatch(threadsActions.refreshStatus());
+		};
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/components/ThreadDetail/index.tsx]: k cần thêm cái này e, chỉ cần loading là dc rồi, load rồi thì có cache show ra

File: apps/mobile/src/app/components/ThreadDetail/index.tsx [hunk 2]
```diff
@@ -42,20 +44,25 @@ export default function CreateThreadModal({ navigation, route }: MenuThreadScree
 
 	const currentChannel = useSelector(selectCurrentChannel);
 	const isThread = checkIsThread(currentChannel as ChannelsEntity);
+	const loadingStatus = useSelector((state: RootState) => state?.threads?.loadingStatus);
+	const isLoading = useMemo(() => ['loading', 'not loaded'].includes(loadingStatus), [loadingStatus]);
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/components/ThreadDetail/index.tsx]: chỉ cần loading thoi

File: libs/store/src/lib/threads/threads.slice.ts [hunk 3]
```diff
@@ -214,6 +214,9 @@ export const threadsSlice = createSlice({
 					}
 				});
 			}
+		},
+		refreshStatus: (state) => {
+			state.loadingStatus = 'not loaded';
```
Comment (phuocnguyenncc) [inline libs/store/src/lib/threads/threads.slice.ts]: can remove

**Comment by loi-huynhphuc** [comment]
Screenshot 2024-11-08 144858

**Review by phuocnguyenncc — APPROVED** [review]

**Pull request merged**